tests/test_post_gen_project.py

```diff
@@ -0,0 +1,35 @@
+import sys
+from pathlib import Path
+
+import pytest
+
+sys.path.append(str(Path(__file__).resolve().parents[1]))
+from hooks.post_gen_project import get_repo_dir  # noqa: E402
+
+
+def test_with_repo_get_repo_dir(monkeypatch, tmp_path):
```

Comment:
This is a good start.

the way we write tests is that we first capture the behavior we want,  described in a comment, then we write tests for that.

Since we are doing a bunch of filesystem interactions, we may want to build a filesystem as a test fixture and use it across multiple tests.  We want to use the pytest tmpdir capabilities which build and break down the temporary files regardless of how the test ends.

The assert should generally look like `assert expected == actual` and above we define what is `expected` and what is `actual`.  This is not so different from what you did, but it just makes things easier to read.